### PR TITLE
Support socket.io version 1.4+

### PR DESCRIPTION
**Pull request by alexgerv**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "express-graceful-exit",
-  "version": "0.3.3",
+  "version": "1.0.0-gogap",
```
Comment (ivolucien) [inline package.json]: I agree that 1.0.0 is overdue, but what is the "-gogap" is about?

File: lib/graceful-exit.js [hunk 2]
```diff
@@ -80,8 +80,10 @@ exports.gracefulExitHandler = function(app, server, _options) {
 
   // Disconnect all the socket.io clients
   if (options.socketio) {
-    options.socketio.sockets.clients().forEach(function(socket) {
-      logger('Killing socketio socket');
+    var connectedSockets = Object.values(options.socketio.sockets.sockets);
+    logger('Disconnecting ' + connectedSockets.length + ' sockets...');
+    options.socketio.close();
+    connectedSockets.forEach(function(socket) {
```
Comment (ivolucien) [inline lib/graceful-exit.js]: Alex, thank you for submitting a patch.

Could you give me some background on what changed with socket.io v1.4 that requires this change? And is it backwards compatible?  I scanned the socket.io change blog and didn't see anything about it.

File: package.json (hunk 1 above)
Comment (alexgerv) [inline package.json]: Sorry, this is a leftover. I'll adjust.

File: lib/graceful-exit.js (hunk 2 above)
Comment (alexgerv) [inline lib/graceful-exit.js]: Changes are most likely not backward compatible, though I haven't check. The major increase to version `1.0.0` should be good for not affecting other people already using this.

**Comment by alexgerv** [comment]
@ivolucien are we good to merge?

By the way I'm interested to collaborate on the repo if you guys are looking.
cc @emostar

**Comment by ivolucien** [comment]
Hi Alex, sorry for the very slow responses.

I would much rather provide at least 1 minor version that deprecates the previous versions of socket.io before dropping support.

I know that some folks are stuck on old versions in production code and (however inadvisedly) don't have proper version limits set. I'll write it in mid-January, or review and merge a PR more promptly.

**Comment by alexgerv** [comment]
Thanks @ivolucien !

**Comment by ivolucien** [comment]
@alexgerv - please test PR #7 with your code base and let me know if you have any feedback.
Thank you for your patience.

**Comment by ivolucien** [comment]
@alexgerv - I've written a more general set of socket.io improvements based on your contribution, thank you!  Hopefully better late than never.

**Pull request closed without merging**